wnfs-wasm/src/fs/utils.rs

```diff
@@ -9,12 +9,19 @@ use wnfs::{
     Metadata,
 };
 
-use super::{metadata::JsMetadata, PrivateDirectory, PrivateForest, PublicDirectory};
+use super::{metadata::JsMetadata, PrivateDirectory, PrivateFile, PrivateForest, PublicDirectory};
 
 //--------------------------------------------------------------------------------------------------
 // Functions
 //--------------------------------------------------------------------------------------------------
 
+pub(crate) fn error<E>(message: &str) -> impl FnOnce(E) -> js_sys::Error + '_
+where
+    E: Debug,
+{
+    move |e| Error::new(&format!("{message}: {e:?}"))
+}
+
 pub(crate) fn map_to_rust_vec<T, F: FnMut(JsValue) -> JsResult<T>>(
```

Comment:
minor: this is not a rust vec at the end though right?

Reply:
nvm, pr didn't give me the proper lines haha.

Reply:
It is a rust vec in the end wrapped in Result.

Reply:
oh i know. i had to go into the file and see what the underlying code was. The PR didn't help on the view haha.